### ISSUE TITLE
Investigate queue behavior when operation not marked as OneWay

### ISSUE THREAD

**Issue by g7ed6e** [CoreWCF/CoreWCF#1046]
> @mconnew Can you review this please ? I still have concern with handling a message that is dispatched successfully but service implementation throws. In this case, `ImmutableDispatchRuntime` is calling `RequestContext.ReplayAsync(null)` which causes `ReceiveContext.CompleteAsync` to be called before the operation is invoked. This behavior leads to a situation where the actual `QueueTransportPump` implementation is not able to know if invocation threw or not.

Does CoreWCF throw if you don't have an operation marked as OneWay? If it's limited to only using OneWay methods, then ReplyAsync should do nothing if it gets passed a null.  

Other than that being resolved, everything else looks good. I feel like that can be deferred for now to unblock you.

_Originally posted by @mconnew in https://github.com/CoreWCF/CoreWCF/pull/1018#issuecomment-1491004768_
